examples/polypharmacy_gripnet/main.py

```diff
@@ -1,73 +1,60 @@
+    device = torch.device(device)
 
     # ---- setup configs ----
     cfg = get_cfg_defaults()
-    cfg.merge_from_file(args.cfg)
     cfg.freeze()
     seed.set_seed(cfg.SOLVER.SEED)
-    # ---- setup logger and output ----
-    output_dir = os.path.join(cfg.OUTPUT_DIR, cfg.DATASET.NAME, args.output)
-    os.makedirs(output_dir, exist_ok=True)
-    logger = lu.construct_logger("gripnet", output_dir)
-    logger.info("Using " + device)
-    logger.info(cfg.dump())
-    # ---- setup dataset ----
-    download_file_by_url(cfg.DATASET.URL, cfg.DATASET.ROOT, "pose.pt", "pt")
-    data = torch.load(os.path.join(cfg.DATASET.ROOT, "pose.pt"))
-    device = torch.device(device)
-    data = data.to(device)
-    # ---- setup model ----
-    print("==> Building model..")
-    model = GripNet(
-        cfg.GRIPN.GG_LAYERS, cfg.GRIPN.GD_LAYERS, cfg.GRIPN.DD_LAYERS, data.n_d_node, data.n_g_node, data.n_dd_edge_type
-    ).to(device)
-    # TODO Visualize model
-    # ---- setup trainers ----
-    optimizer = torch.optim.Adam(model.parameters(), lr=cfg.SOLVER.BASE_LR)
-    # TODO
-    trainer = Trainer(cfg, device, data, model, optimizer, logger, output_dir)
-
-    if args.resume:
-        # Load checkpoint
-        print("==> Resuming from checkpoint..")
-        cp = torch.load(args.resume)
-        trainer.model.load_state_dict(cp["net"])
-        trainer.optim.load_state_dict(cp["optim"])
-        trainer.epochs = cp["epoch"]
-        trainer.train_auprc = cp["train_auprc"]
-        trainer.valid_auprc = cp["valid_auprc"]
-        trainer.train_auroc = cp["train_auroc"]
-        trainer.valid_auroc = cp["valid_auroc"]
-        trainer.train_ap = cp["train_ap"]
-        trainer.valid_ap = cp["valid_ap"]
-
-    trainer.train()
+
+    # ---- setup dataset and data loader ----
+    data = load_data(cfg.DATASET)
+    dataloader_train, _ = get_all_dataloader(data)
+
+    # ---- setup supergraph ----
+    # create gene and drug supervertex
+    supervertex_gene = SuperVertex("gene", data.g_feat, data.gg_edge_index)
+    supervertex_drug = SuperVertex("drug", data.d_feat, data.train_idx, data.train_et)
+
+    # create superedge form gene to drug supervertex
+    superedge = SuperEdge("gene", "drug", data.gd_edge_index)
+
+    setting_gene = setup_supervertex(cfg.GRIPN_SV1)
+    setting_drug = setup_supervertex(cfg.GRIPN_SV2)
+
+    # construct supergraph
+    supergraph = SuperGraph([supervertex_gene, supervertex_drug], [superedge])
+    supergraph.set_supergraph_para_setting([setting_gene, setting_drug])
+
+    # ---- setup model and trainer ----
+    model = GripNetLinkPrediction(supergraph, cfg.SOLVER)
+    print(model)
+
+    trainer = pl.Trainer(
+        default_root_dir=cfg.OUTPUT_DIR,
+        max_epochs=cfg.SOLVER.MAX_EPOCHS,
+        log_every_n_steps=cfg.SOLVER.LOG_EVERY_N_STEPS,
+    )
+
+    # ---- train and test ----
+    trainer.fit(model, dataloader_train)
+    test_result = trainer.test(model, dataloader_train)
 
```

Comment:
Clarify the test results are not on Test set?

Reply:
done

Reply:
1. a bit more explanation on why will help users.
2. Why break new line when it can fit 120? Are you using 80 max length? Please ensure style consistency.

Reply:
Better to tell users what issue here for clarity. Need to talk.

Reply:
Or in the ReadMe.